### PR TITLE
Zb/models handlers routes

### PR DESCRIPTION
**Pull request by zbo14**
This PR piggybacks on the noice refactoring job you did and does a couple things...

* Breaks out models, handlers, and routes
* Returns better validation error messages in responses
* Removes docs in comments and uses [hapi-swagger](https://github.com/glennjones/hapi-swagger) to auto-generate api docs from c0de (when you `npm start` you can view the documentation at `localhost:9000/documentation`)
* Adds a lil to the db module and a test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,9 +3,7 @@
   "version": "1.0.0",
   "scripts": {
     "test": "nyc mocha --reporter spec --recursive",
-    "start": "nodemon server.js",
-    "doc:build": "swagger-jsdoc -d swaggerDef.js ./lib/routes/*.js ./lib/schemas/*.js -o swagger.json",
-    "doc:serve": "redoc-cli serve swagger.json"
+    "start": "NODE_ENV=development nodemon main.js"
```
Comment (ngriffiths21) [inline package.json]: Annoyingly, this doesn't work on windows. It has to be "set NODE_ENV..."

File: lib/models.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const db = require('./db')
+
+module.exports = name => {
```
Comment (ngriffiths21) [inline lib/models.js]: This is cool!

File: main.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  })
+
+  routes(server)
+
+  if (process.env.NODE_ENV === 'development') {
+    await server.register([
+      require('inert'),
+      require('vision'),
+      {
+        plugin: require('hapi-swagger'),
```
Comment (ngriffiths21) [inline main.js]: This makes it way easier, I like it, but can we also document status codes/response schema?

Comment (zbo14) [reply main.js]: yeah, check [this](https://github.com/glennjones/hapi-swagger/blob/master/usageguide.md#status-codes) out!

**Comment by ngriffiths21** [comment]
merge whenever you're ready

**Pull request merged**